packages/pds/src/account-manager/oauth-store.ts

```diff
@@ -0,0 +1,483 @@
+} from '@atproto/xrpc-server'
+import { ActorStore } from '../actor-store/actor-store'
+import { BackgroundQueue } from '../background'
+import { ImageUrlBuilder } from '../image/image-url-builder'
+import { ServerMailer } from '../mailer'
+import { Sequencer } from '../sequencer'
+import { AccountManager } from './account-manager'
+import { AccountStatus, ActorAccount } from './helpers/account'
+import * as authRequest from './helpers/authorization-request'
+import * as device from './helpers/device'
+import * as deviceAccount from './helpers/device-account'
+import * as token from './helpers/token'
+import * as usedRefreshToken from './helpers/used-refresh-token'
+
+/**
+ * This class' purpose is to implement the interface needed by the OAuthProvider
+ * to interact with the account database (through the {@link AccountManager}).
+ *
+ * @note The use of this class assumes that there is no entryway.
+ */
+export class OAuthStore
+  implements AccountStore, RequestStore, DeviceStore, TokenStore
+{
+  constructor(
+    private readonly accountManager: AccountManager,
+    private readonly actorStore: ActorStore,
+    private readonly imageUrlBuilder: ImageUrlBuilder,
+    private readonly backgroundQueue: BackgroundQueue,
+    private readonly mailer: ServerMailer,
+    private readonly sequencer: Sequencer,
+    private readonly plcClient: Client,
+    private readonly plcRotationKey: Keypair,
+    private readonly publicUrl: string,
+    private readonly recoveryDidKey: string | null,
+  ) {}
+
+  private get db() {
+    const { db } = this.accountManager
+    if (db.destroyed) throw new Error('Database connection is closed')
+    return db
+  }
+
+  private get serviceDid() {
+    return this.accountManager.serviceDid
+  }
+
+  private async buildAccount(row: Selectable<ActorAccount>): Promise<Account> {
+    const account = deviceAccount.toAccount(row, this.serviceDid)
+
+    if (!account.name || !account.picture) {
+      const did = account.sub
+
+      const profile = await this.actorStore.read(did, async (store) => {
+        return store.record.getProfileRecord()
+      })
+
+      if (profile) {
+        const { avatar, displayName } = profile
+
+        account.name ||= displayName
+        account.picture ||= avatar
+          ? this.imageUrlBuilder.build('avatar', did, avatar.ref.toString())
+          : undefined
+      }
+    }
+
+    return account
+  }
+
+  private async verifyEmailAvailability(email: string): Promise<void> {
+    // @NOTE Email validity & disposability check performed by the OAuthProvider
+
+    const account = await this.accountManager.getAccountByEmail(email, {
+      includeDeactivated: true,
+      includeTakenDown: true,
+    })
+
+    if (account) {
+      throw new InvalidRequestError(`Email already taken: ${email}`)
+    }
+  }
+
+  // AccountStore
+
+  async createAccount({
+    inviteCode,
+    handle,
+    email,
+    password,
+    // @TODO: where is the birthdate saved ?
```

Comment:
I believe it is usually kept in the special-cased app user preferences.

Reply:
I'll remove it from the PDS then, as there is no place to store this data in the PDS until we have private state